shell/common/resource_cache_limit_calculator.cc

```diff
@@ -0,0 +1,27 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/common/resource_cache_limit_calculator.h"
+
+namespace flutter {
+
+size_t ResourceCacheLimitCalculator::GetResourceCacheMaxBytes() {
+  size_t max_bytes = 0;
+  size_t max_bytes_threshold = max_bytes_threshold_ > 0
+                                   ? max_bytes_threshold_
+                                   : std::numeric_limits<size_t>::max();
+  auto it = items_.begin();
+  while (it != items_.end()) {
+    fml::WeakPtr<ResourceCacheLimitItem> item = *it;
+    if (item) {
+      max_bytes += item->GetResourceCacheLimit();
+      ++it;
+    } else {
+      it = items_.erase(it);
```

Comment:
This is `O(n)` per erase.

We could do better with something like

```cpp
std::vector live_items;
for (auto item : items_) {
  if (item) {
    live_items.push_back(item);
    max_bytes += item->GetResourceCacheLimit();
  }
}
items_= std::move(new_items);
```

Reply:
done